lib/rules/no-duplicate-imports.js

```diff
@@ -21,18 +21,107 @@ function getValue(node) {
     return "";
 }
 
+/**
+ * Checks if the import's type is the given type import.
+ * @param {ASTNode} node The import to be checked.
+ * @param {type} type the type to be checked.
+ *
+ * @returns {boolean} Whether or not the import's type is type.
+ */
+function checkImportType(node, type) {
+    return (
+        node.specifiers &&
+        node.specifiers[0] &&
+        node.specifiers[0].type === type
+    );
+}
+
+
+/**
+ * Checks whether a node has the given value.
+ * @param {string} value the value to be checked.
+ *
+ * @returns {Function} the predicate function, return true if the parameter ASTNode has the given value.
+ */
+function hasValue(value) {
+    return function(node) {
+        return getValue(node) === value;
+    };
+}
+
+/**
+ * Checks if two nodes can be merged.
+ * @param {ASTNode} node1 one of the two nodes to be checked.
+ * @param {ASTNode} node2 the other node to be checked.
+ *
+ * @returns {boolean} true where only one ImportNamespaceSpecifier and one ImportSpecifier, otherwise, false.
+ */
+function isTwoNodesCanMerge(node1, node2) {
+    return (
+        (
+            checkImportType(node1, "ImportNamespaceSpecifier") &&
+            checkImportType(node2, "ImportSpecifier")
+        ) ||
+        (
+            checkImportType(node1, "ImportSpecifier") &&
+            checkImportType(node2, "ImportNamespaceSpecifier")
+        )
+    );
+}
```

Comment:
Looks like this function is actually doing the opposite of its name: checks if two nodes _cannot_ be merged?